util/azure/azure.go

```diff
@@ -17,28 +17,12 @@ limitations under the License.
 package azure
 
 import (
-	"fmt"
-	"regexp"
 	"strings"
 )
 
-var azureResourceGroupNameRE = regexp.MustCompile(`.*/subscriptions/(?:.*)/resourceGroups/(.+)/providers/(?:.*)`)
-
 // AzureSystemNodeLabelPrefix is a standard node label prefix for Azure features, e.g., kubernetes.azure.com/scalesetpriority.
 const AzureSystemNodeLabelPrefix = "kubernetes.azure.com"
 
-// ConvertResourceGroupNameToLower converts the resource group name in the resource ID to be lowered.
-// Inspired by https://github.com/kubernetes-sigs/cloud-provider-azure/blob/88c9b89611e7c1fcbd39266928cce8406eb0e728/pkg/provider/azure_wrap.go#L409
-func ConvertResourceGroupNameToLower(resourceID string) (string, error) {
```

Comment:
I wonder if we want to add `// Deprecated :` marker here and then remove this function in next minor release

Reply:
My vote: let's just remove it. We don't make any guarantees on using CAPZ as a library and if anyone else happened to be using this function they can just use the cloud-provider-azure one instead.